builtin/providers/cloudstack/resource_cloudstack_ipaddress.go

```diff
@@ -138,10 +153,21 @@ func resourceCloudStackIPAddressDelete(d *schema.ResourceData, meta interface{})
 func verifyIPAddressParams(d *schema.ResourceData) error {
 	_, network := d.GetOk("network_id")
 	_, vpc := d.GetOk("vpc_id")
+	_, zone := d.GetOk("zone_id")
+
+	f := func(bs ...bool) int {
+		cnt := 0
+		for _, b := range bs {
+			if b {
+				cnt++
+			}
+		}
+		return cnt
+	}
 
-	if (network && vpc) || (!network && !vpc) {
+	if f(network, vpc, zone) >= 2 || (!network && !vpc && !zone) {
 		return fmt.Errorf(
-			"You must supply a value for either (so not both) the 'network_id' or 'vpc_id' parameter")
+			"You must supply one value for either (so not more than two) the 'network_id' or 'vpc_id' or 'zone_id'  parameter")
```

Comment:
ok, I confirmed that it allows pass both zone_id and network_id together. 

```
cloudstack_ipaddress.ip-vm01: Creating...
  ip_address: "" => "<computed>"
  network_id: "" => "xxxxxxxxxxxxxxxxxx"
  project:    "" => "<computed>"
  zone_id:    "" => "xxxxxxxxxxxxxxxxxxx"
cloudstack_ipaddress.ip-vm01: Creation complete

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

I will restore few functions.